Handle array export and revise call wrapper (#36)
InverseIntegral/js2wasm: README.md

# Usage
In order to compile a function you have to supply it as a string to the `transpile` function.
The returned value is of the type CallWrapper.

```javascript
const wrapper = new Transpiler().transpile('function add(a, b) { return a + b; }');
wrapper.setFunctionName('add').call(21, 21);
```

# API

> ## Transpiler creation

Reviewer: Is it necessary to distinguish between the creation and the manipulation of the transpiler?

> * new Transpiler(): `void` <br />

Reviewer: Why is only the return type marked as code?

Author: I don't think it looks that good if the entire line is marked as code.

  Creates a new transpiler with an empty hook class.
* new Transpiler(hooks: `TranspilerHooks`): `void` <br />
  Creates a new transpiler with the specified hook instance.

## Transpiler manipulation
* Transpiler#**transpile**(content: `string`): `CallWrapper` <br />
  Transpiles the specified content.

## CallWrapper manipulation
* CallWrapper#**setFunctionName**(functionName: `string`): `CallWrapper` <br />
  Set the name of the initial function to be called.
* CallWrapper#**call**(...parameters: `any[]`): `any` <br />

>   Call the initial function of the transpiled Wasm code.

Reviewer: ```suggestion
  Calls the initial function of the transpiled Wasm code.
```

## TranspilerHooks manipulation

> All methods are hook methods, which are called by the `Transpiler` and the `CallWrapper`. The default implementation of all these hook methods is empty, but it can be derived from and passed along to the `Transpiler`.

Reviewer: ```suggestion
The following methods are hook methods, which get called by the `Transpiler` and the `CallWrapper`. The default implementations of them are empty and can be overwritten.
```

Author: I had to change this a little bit, because `TranspilerHooks` now is an interface.

* TranspilerHooks#**beforeCompilation**(): `void` <br />

>   Called before the transpilation process

Reviewer: Add full stops after those sentences.
